pulser-core/pulser/math/abstract_array.py

```diff
@@ -37,19 +37,27 @@ class AbstractArray:
         array: The array to store.
     """
 
-    def __init__(self, array: AbstractArrayLike, dtype: DTypeLike = None):
+    def __init__(
+        self,
+        array: AbstractArrayLike,
+        dtype: DTypeLike = None,
+        force_array: bool = False,
+    ):
         """Initializes a new AbstractArray."""
         self._array: np.ndarray | torch.Tensor
         if isinstance(array, AbstractArray):
             self._array = array._array
-        elif self.has_torch() and type(array) is torch.Tensor:
+        elif self.has_torch() and isinstance(array, torch.Tensor):
             self._array = torch.as_tensor(
                 array,
                 dtype=dtype,  # type: ignore[arg-type]
             )
         else:
             self._array = np.asarray(array, dtype=dtype)
 
+        if force_array:
+            self._array = self._array[None]
```

Comment:
We only want to do this for scalars:
```suggestion
        if force_array and self._array.ndim == 0:
            self._array = self._array[None]
```